### PR TITLE
Add medic flag taking

### PR DESCRIPTION
**Pull request by 165135**
Adds clicking flag with bandage takes/captures a flag.
(I wouldn't be surprised if this is a wrong and inefficient way to do it)

### CODE REVIEW

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua [hunk 1]
```diff
@@ -9,72 +9,84 @@ function ctf_healing.register_bandage(name, def)
...
+			    end
+
+			    local hp_add = math.random(def.heal_min or 3, def.heal_max or 4)
+
+			    if hp + hp_add > limit then
+				    hp_add = limit - hp
+				    hp = limit
+			    else
+				    hp = hp + hp_add
+			    end
```
Comment (farooqkz) [inline mods/ctf/ctf_combat/ctf_healing/bandage.lua:59]:
```suggestion
			    hp = math.min(hp + hp_add, limit)
```

**Review by farooqkz — CHANGES_REQUESTED** [review]

**Comment by farooqkz** [comment]
@LoneWolfHT  LGTM

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua (hunk 1 above)
Comment (165135) [reply mods/ctf/ctf_combat/ctf_healing/bandage.lua:59]: I didn't really mess with this part but sure

**Comment by farooqkz** [comment]
@165135 As I said somewhere else, you could simply commit my suggestion from Github UI rather than making your own commit.

**Comment by 165135** [comment]
Wasn't sure, got it :+1:

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua [hunk 2]
```diff
@@ -9,72 +9,79 @@ function ctf_healing.register_bandage(name, def)
...
+					    color = "warning",
+				    })
+				    return
+			    end
+
+			    local hp_add = math.random(def.heal_min or 3, def.heal_max or 4)
+
+			    hp = math.min(hp + hp_add, limit)
+
+			    local result = RunCallbacks(ctf_healing.registered_on_heal, player, object, hp_add)
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_combat/ctf_healing/bandage.lua:56]: hp_add isn't always accurate when passed to the registered_on_heal callbacks

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua [hunk 3]
```diff
@@ -9,72 +9,79 @@ function ctf_healing.register_bandage(name, def)
...
+				    hud_events.new(uname, {
+					    quick = true,
+					    text = result,
+					    color = "warning",
+				    })
+                end
+	        elseif pointed_thing.type == "node" then
+                local node_pointed = minetest.get_node(pointed_thing.under)
+                local node_above_pos = {x=pointed_thing.under.x, y=pointed_thing.under.y +1, z=pointed_thing.under.z}
+                local node_above = minetest.get_node(node_above_pos)
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_combat/ctf_healing/bandage.lua:75]:
```suggestion
                local node_pointed = minetest.get_node(pointed_thing.under)
                local node_above = minetest.get_node(pointed_thing.under:offset(0, 1, 0))
```

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua (hunk 2 above)
Comment (165135) [reply mods/ctf/ctf_combat/ctf_healing/bandage.lua:56]: image
I didn't change the bandage healing code except for this edit that @farooqkz suggested

Comment (LoneWolfHT) [reply mods/ctf/ctf_combat/ctf_healing/bandage.lua:56]: @farooqkz Can you see what functionality you removed with that suggestion?

Comment (165135) [reply mods/ctf/ctf_combat/ctf_healing/bandage.lua:56]: @LoneWolfHT I could just put it back the way it was and not worry about it

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua [hunk 4]
```diff
@@ -9,72 +9,83 @@ function ctf_healing.register_bandage(name, def)
...
+                end
+	        elseif pointed_thing.type == "node" then
+                local node_pointed = minetest.get_node(pointed_thing.under)
+                local node_above = minetest.get_node(pointed_thing.under:offset(0, 1, 0))
+                if node_pointed.name ~= "ctf_modebase:flag_captured_top" then
+                    if node_pointed.name:find("ctf_modebase:flag_") then
+		                ctf_modebase.flag_on_punch(player, pointed_thing.under, node_pointed)
+                    elseif node_above.name:find("ctf_modebase:flag_") then
+                        ctf_modebase.flag_on_punch(player, pointed_thing.under:offset(0, 1, 0), node_above)
+                    end
```
Comment (farooqkz) [inline mods/ctf/ctf_combat/ctf_healing/bandage.lua:85]:
```suggestion
                local node_pointed = minetest.get_node(pointed_thing.under)
                local node_above = minetest.get_node(pointed_thing.under:offset(0, 1, 0))
                local flag_captured_top = "ctf_modebase:flag_captured_top"
                if node_pointed.name ~= "ctf_modebase:flag_captured_top" then
                    if node_pointed.name:find("ctf_modebase:flag_") then
		                ctf_modebase.flag_on_punch(player, pointed_thing.under, node_pointed)
                    elseif node_above.name:find("ctf_modebase:flag_") and node_above.name ~= flag_captured_top then
                        ctf_modebase.flag_on_punch(player, pointed_thing.under:offset(0, 1, 0), node_above)
                    end
```

It was crashing when the flag was taken and the bottom was clicked with medic's bandage. This fixes it.

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua (hunk 2 above)
Comment (LoneWolfHT) [reply mods/ctf/ctf_combat/ctf_healing/bandage.lua:61]: :+1: It's just something they could learn a little from

File: mods/ctf/ctf_combat/ctf_healing/bandage.lua [hunk 5]
```diff
@@ -9,72 +9,84 @@ function ctf_healing.register_bandage(name, def)
...
+				    hud_events.new(uname, {
+					    quick = true,
+					    text = result,
+					    color = "warning",
+				    })
+                end
+	        elseif pointed_thing.type == "node" then
+                local node_pointed = minetest.get_node(pointed_thing.under)
+                local node_above = minetest.get_node(pointed_thing.under:offset(0, 1, 0))
+                local flag_captured_top = "ctf_modebase:flag_captured_top"
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_combat/ctf_healing/bandage.lua:80]: This variable isn't needed

Comment (165135) [reply mods/ctf/ctf_combat/ctf_healing/bandage.lua]: Removed ``flag_captured_top``, I think that's what you meant

**Pull request merged**